packages/lumx-react/src/components/date-picker/DatePickerControlled.tsx

```diff
@@ -21,6 +21,10 @@ export interface DatePickerControlledProps extends DatePickerProps {
     onPrevMonthChange(): void;
     /** On next month change callback. */
     onNextMonthChange(): void;
+    /** Set offset for the month */
+    setMonthOffset: (value: React.SetStateAction<number>) => void;
+    /** Current offset for month */
+    monthOffset: number;
```

Comment:
I think we need to rework this. 
The `selectedMonth` is sufficient to know which month/year to display, no need for `monthOffset`. The `monthOffset` is an internal implementation detail of the DatePicker that I think we should change.

Here, I think we should do:
```suggestion
    /** On month/year change callback. */
    onMonthChange?: (newMonth: Date) => void;
```
Note the prop is optional because `DatePickerControlled` is exposed to the public and we cannot add required props without a breaking change.

And then in DatePicker we should do:
```ts
    const [selectedMonth, setSelectedMont] = useState(referenceDate);
    const setPrevMonth = () => setMonthOffset(current => addMonthResetDay(current, -1));
    const setPrevMonth = () => setMonthOffset(current => addMonthResetDay(current, +1));
    const onMonthChange = setSelectedMont;
```

Reply:
You're right, now that you mention it, if the onMonthChange is optional, the year field won't work properly for users that didn't set it. Shouldn't we display the field only if onMonthChange is defined, and let the text version for users that didn't pass a onMonthChange?

Reply:
fixed